ocf_datapipes/transform/numpy/sun_position.py

```diff
@@ -0,0 +1,104 @@
+import warnings
+
+import numpy as np
+import pandas as pd
+import pvlib
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+
+from ocf_datapipes.utils.consts import BatchKey
+from ocf_datapipes.utils.geospatial import osgb_to_lat_lon
+
+ELEVATION_MEAN = 37.4
+ELEVATION_STD = 12.7
+AZIMUTH_MEAN = 177.7
+AZIMUTH_STD = 41.7
+
+
+@functional_datapipe("add_sun_position")
+class AddSunPositionIterDataPipe(IterDataPipe):
+    def __init__(self, source_dp: IterDataPipe, modality_name: str):
+        self.source_dp = source_dp
+        self.modality_name = modality_name
+        assert self.modality_name in ["hrvsatellite", "gsp", "gsp_5_min", "pv", "nwp_target_time"]
+
+    def __iter__(self):
```

Comment:
can we break this up a bit, so its siltyly easier to read

Reply:
Yeah, I can do that